types/chart.js/index.d.ts

```diff
@@ -157,7 +157,20 @@ declare namespace Chart {
 
     type PositionType = 'left' | 'right' | 'top' | 'bottom' | 'chartArea';
 
-    type InteractionMode = 'point' | 'nearest' | 'single' | 'label' | 'index' | 'x-axis' | 'dataset' | 'x' | 'y';
+    // Allow extending the IteractionMode type alias
+    // see https://github.com/microsoft/TypeScript/issues/28078#issuecomment-432339564
+    interface InteractionModeRegistry {
+        'point': 'point';
+        'nearest': 'nearest';
+        'single': 'single';
+        'label': 'label';
+        'index': 'index';
+        'xaxis': 'xaxis';
```

Comment:
Why did you change this from `x-axis` to `xaxis`?

Reply:
You are right, this was a mistake. Will fix it soon